src/Data/MessagePack/Class.hs

```diff
@@ -106,8 +106,9 @@ instance MessagePack Object where
   fromObject = return
 
 instance MessagePack () where
-  toObject _ = ObjectArray []
+  toObject _ = ObjectNil
   fromObject = \case
+    ObjectNil      -> return ()
     ObjectArray [] -> return ()
```

Comment:
Actually, I thought decoding empty arrays to () should be fine. If it's not, it's perfectly fine to remove that.